iceoryx_dust/vocabulary/include/iox/detail/span_iterator.hpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2022 - 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_VOCABULARY_SPAN_ITERATOR_HPP
+#define IOX_DUST_VOCABULARY_SPAN_ITERATOR_HPP
+
+// Use 'assert's as 'iox::Expects' is not useable inside 'constexpr' functions
+#include <cassert>
+#include <iterator> // for reverse_iterator, distance, random_access_...
+
+namespace iox
+{
+
+template <typename T>
+class span_iterator
+{
+  public:
+    using iterator_category = std::random_access_iterator_tag;
+    using value_type = std::remove_cv_t<T>;
+    using difference_type = std::ptrdiff_t;
+    using pointer = T*;
+    using reference = T&;
+
+    constexpr span_iterator() = default;
```

Comment:
Is there a use case for a defaulted span_iterator?

Reply:
I can't think of any use-cases, removed.